app/org/sagebionetworks/bridge/stormpath/StormpathAccountDao.java

```diff
@@ -80,6 +81,13 @@ public void setStudyService(StudyService studyService) {
         this.studyService = studyService;
     }
     
+    public StormpathAccountDao() {
+        BridgeConfig config = BridgeConfigFactory.getConfig();
+        String url = config.getStormpathApplicationHref().trim();
+        String[] parts = url.split("/");
+        applicationId = parts[parts.length-1];
+    }
```

Comment:
[POLISH] It would be much cleaner and much more testable to have the StormpathAccountDao take in the applicationId (either in the constructor or as a setter) and have the Spring bean configure it. This way, we don't have to hit production configs to run our tests.